src/git-utils.stanza

```diff
@@ -3,10 +3,48 @@ defpackage slm/git-utils:
   import collections
 
   import maybe-utils
+  import semver
 
   import slm/file-utils
   import slm/process-utils
 
+defn parse-git-version (msg:String) -> Maybe<SemanticVersion> :
+  val comps = split(msg, " ")
+  for comp in comps first:
+    parse-semver(comp)
+
+defn run-git-version () -> Maybe<SemanticVersion> :
+  try:
+    val p = ProcessBuilder(["git", "version"])
+      $> with-output
+      $> build
+
+    ; Evidently - you have to call `get-output` before `run` here
```

Comment:
This is not relevant to this PR but just looked over the `ProcessBuilder` code and it uses the `Process` API a bit awkwardly. Might be worth revamping at some point. The issue is that `run-and-get-exit-code` calls `wait`, which in turn destroys all the process' resources.

Reply:
or we should revamp the `Process` API....